docs/IE_DG/InferenceEngine_QueryAPI.md

```diff
@@ -1,82 +1,238 @@
+* [ie_api.ieCore.get_metric](api/ie_python_api/_autosummary/openvino.inference_engine.IECore.html#openvino.inference_engine.IECore.get_metric) - Provides information about specific device.
+* [ie_api.IECore.get_config](api/ie_python_api/_autosummary/openvino.inference_engine.IECore.html#openvino.inference_engine.IECore.get_config) - Gets the current value of a specific configuration key.
+* [ie_api.IECore.set_config](api/ie_python_api/_autosummary/openvino.inference_engine.IECore.html#openvino.inference_engine.IECore.set_config)  - Sets a new value for the configuration key.
+
+The [ie_api.ExecutableNetwork](api/ie_python_api/_autosummary/openvino.inference_engine.ExecutableNetwork.html) class is also extended to support the Query API:
+* [ie_api.ExecutableNetwork.get_metric](api/ie_python_api/_autosummary/openvino.inference_engine.ExecutableNetwork.html#openvino.inference_engine.ExecutableNetwork.get_metric)
+* [ie_api.ExecutableNetwork.get_config](latest/api/ie_python_api/_autosummary/openvino.inference_engine.ExecutableNetwork.html#openvino.inference_engine.ExecutableNetwork.get_config)
+* There is no method to call for set_config, but the equivalent action is described below.
+
+### Query API in the IECore Class
+
+#### Get Available Devices
+
+```python
+from openvino.inference_engine import IECore
+
+ie = IECore()
+print(ie.available_devices)
+```
+
+This code prints a list of available devices, for example:
+
+```
+MYRIAD.1.2-ma2480
+MYRIAD.1.4-ma2480
+FPGA.0
+FPGA.1
+CPU
+GPU.0
+GPU.1
+```
+
+Each device name can then be passed to:
+
+* `IECore.load_network` to load the network to a specific device.
+* `IECore.get_metric` to get common or device specific metrics.
+* All other methods of the `IECore` class that accept `deviceName`.
+
+#### Get Metric
+
+To extract device properties such as available device, device name, supported configuration keys, and others, use the [IECore.get_metric](api/ie_python_api/_autosummary/openvino.inference_engine.IECore.html#openvino.inference_engine.IECore.get_metric) method:
+
+```python
+from openvino.inference_engine import IECore
+
+ie = IECore()
+ie.get_metric(device_name="CPU", metric_name="FULL_DEVICE_NAME")
+```
+
+A returned value appears as follows: `Intel(R) Core(TM) i7-8700 CPU @ 3.20GHz`.
+
+To list all supported metrics for a device:
+
+```python
+from openvino.inference_engine import IECore
+
+ie = IECore()
+ie.get_metric(device_name="GPU", metric_name="SUPPORTED_METRICS")
+```
+
+> **NOTE**: All metrics have a specific type, which is set during the metric instantiation. The list of common device-agnostic metrics can be found in ie_plugin_config.hpp. Device specific metrics (for example, for HDDL, MYRIAD devices) can be found in corresponding plugin folders.
```

Comment:
This note is not relevant for Python developers and adds confustion. My suggestion is to remove it, but the minimum improvement would be to link to the source files.

Reply:
Removed